Plugins/TGeo/src/TGeoLayerBuilder.cpp

```diff
@@ -94,18 +94,47 @@ void Acts::TGeoLayerBuilder::buildLayers(const GeometryContext& gctx,
 
   // Helper function to fill the layer
   auto fillLayer = [&](const LayerSurfaceVector lSurfaces,
-                       const LayerConfig& lCfg) -> void {
-    // Set binning by hand if nb0 > 0 and nb1 > 0
-    auto nb0 = std::get<int>(lCfg.binning0);
-    auto nb1 = std::get<int>(lCfg.binning1);
-    // Or use the binning type
-    auto nt0 = std::get<BinningType>(lCfg.binning0);
-    auto nt1 = std::get<BinningType>(lCfg.binning1);
+                       const LayerConfig& lCfg,
+                       unsigned int pl_id = 0) -> void {
+    int nb0 = 0, nt0 = 0;
+    if (lCfg.binning0.size() <= pl_id) {
+      // Has not been configured for auto-binning
+      if (lCfg.binning0.empty() or (std::get<int>(lCfg.binning0.at(0)) > 0)) {
```

Comment:
I don't quite understand this logic. Is this trying to allow partially auto-binning the outer layers while manually binning the inner ones? I would probably just say that you either have to specify bins for all layers or use auto-binning for all of them.

Reply:
Well yes, if you provide too few configurations because you guessed the wrong number of protolayers, it will try to autobin the rest while issuing a warning. It was actually not meant as a feature, but I can set a misconfiguration to trigger autobinning for all layers and check for all valid configuration to not contain any zeros. Or should this become a failure outright?

Reply:
I would probably make this a failure, so that you can correct your number of configs. Ultimately up to you though.

Reply:
Done. I left the option to autobin only one of the two loc coordinates though, since I am using it. Don't know if we are expecting problems from that?

Reply:
No, that sounds fine to me. Thanks @niermann999!